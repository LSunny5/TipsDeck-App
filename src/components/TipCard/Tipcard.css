.tipLink {
    text-decoration: none;
    color: black;
}

.tipLink:hover {
    color: orange;
}

.cardName {
    font-size: 18px;
    padding: 1rem;
    margin: 0.5rem auto;
    width: 95%;
    background: #82CAFF;
    border-radius: 30px;
    color: black;
    font-weight: bold;
}

/* for tablets */
@media only screen and (min-width: 600px) {
    .cardName {
        padding: 1.8rem;
        margin: 1rem auto;
    }
    
}

/* for large screens */
@media only screen and (min-width: 1024px) {
    .tipLink {
        width: 100%;
    }
}