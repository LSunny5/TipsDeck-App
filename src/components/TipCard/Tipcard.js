import React from 'react';
import './Tipcard.css';
import { Link } from 'react-router-dom';
import TipsDeckContext from '../../TipsDeckContext'
import { getCategoryName } from '../../TipsDeckHelpers';

class Tipcard extends React.Component {
    static contextType = TipsDeckContext;

    render() {
        const {categories = [] } = this.context;
        const { id, name, cat } = this.props;

        const temp = getCategoryName(categories, cat) || {};

        return (
            <div className="Tipbox">
                <Link 
                    to={`/Category/${temp.name}/${id}`} 
                    className="tipLink"
                >
                    <p className="cardName">
                        {name}
                    </p>
                </Link>
            </div>
        )
    }
}

export default Tipcard;