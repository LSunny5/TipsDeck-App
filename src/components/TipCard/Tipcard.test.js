import React from 'react';
import ReactDOM from 'react-dom';
import Tipcard from './Tipcard';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`Tipcard component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Tipcard />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the Tipcard component', () => {
        const wrapper = shallow(<Tipcard />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});
