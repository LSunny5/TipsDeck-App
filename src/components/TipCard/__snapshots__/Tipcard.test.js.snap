// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tipcard component renders the Tipcard component 1`] = `
<div
  className="Tipbox"
>
  <Link
    className="tipLink"
    to="/Category/undefined/undefined"
  >
    <p
      className="cardName"
    />
  </Link>
</div>
`;
